compose/snippets/src/main/java/com/example/compose/snippets/graphics/ShapesSnippets.kt

```diff
@@ -372,12 +373,14 @@ class RoundedPolygonShape(
 fun ApplyPolygonAsClipBasic() {
     // [START android_compose_shapes_apply_as_clip]
     val hexagon = remember {
-        RoundedPolygon(
-            6,
-            rounding = CornerRounding(0.2f)
+        mutableStateOf(
+            RoundedPolygon(
+                6,
+                rounding = CornerRounding(0.2f)
+            )
         )
     }
-    val clip = remember(hexagon) {
+    val clip = remember {
```

Comment:
I don't think your clip will update with a new shape without the key on the remember